docs/source/projects/architecture/content-architecture.rst

```diff
@@ -49,6 +49,7 @@ between the frontends and the IPython Kernel is described in
 The core execution machinery for the kernel is shared with terminal IPython:
 
 .. image:: figs/ipy_kernel_and_terminal.png
+   :alt: Venn diagram showing that Terminal IPython and the IPython Kernel both share the same Python executable. Terminal IPython communicates with the Python executable using stdin and stdout, whereas the Python Kernel uses JSON messages sent over ZeroMQ. 
```

Comment:
Is there a best practice for including information as alt text when that information would *also* be useful as "regular" text that comes just before or after the image? Some of these alt-texts actually read like captions which would be useful regardless of whether the reader is using assistive technology. If we defined these as captions instead of `alt-text` (or used very simple alt-text in addition to captions) would that be an option from an a11y perspective?

Reply:
This is a great question! I've been having this discussion with a few different people recently, but I'm not aware of any best practices written for this yet since what is or isn't  included *outside* of the image is often considered the realm of the documentation's writing style guide (or lack thereof). 

There are, however, resources on how to combine captions and alt text that we could use if we go that direction. Basically, if we'd rather have this as body text or a caption and not alt text, we want to have an empty `alt` element for the images (which means they are skipped). Here's the [W3C recommendation on images with adjacent text](https://www.w3.org/WAI/tutorials/images/decorative/#image-with-adjacent-text-alternative).

There's also been some discussion around whether we want to use [`aria-describedby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) to indicate that there is an image while explicitly linking to its explanation. This could be helpful if the related information is not directly next to the image. I personally think this could be a misuse of the tag, though, especially when our goal is to use proper HTML over adding ARIA. 

cc @tonyfast, @gabalafou, and @ohrely (in case you all are interested)

---

Overall, I'm open to making some of these alt text contributions captions of some kind in this or a future PR. My main goal is to give these sprint contributors credit (via commit) however we decide to do so.

Reply:
Hmmm - here's an idea: what if we converted these into figures with captions, instead of images. [Here's an example of what they would look like from Jupyter Book](https://jupyterbook.org/content/figures.html?highlight=figures#figures).

There's a directive for this in Sphinx. Basically, this image directive:

```rst
.. image:: figs/ipy_kernel_and_terminal.png
   :alt: Venn diagram showing that Terminal IPython and the IPython Kernel both share the same Python executable. Terminal IPython communicates with the Python executable using stdin and stdout, whereas the Python Kernel uses JSON messages sent over ZeroMQ. 
```

would become this figure directive:

```rst
.. figure:: figs/ipy_kernel_and_terminal.png
   
   Venn diagram showing that Terminal IPython and the IPython Kernel both share the same Python executable. Terminal IPython communicates with the Python executable using stdin and stdout, whereas the Python Kernel uses JSON messages sent over ZeroMQ. 
```

Reply:
Here's an example from the Jupyter Book docs:

https://jupyterbook.org/en/stable/content/figures.html#figures

Here's a sample HTML snippet:

```html
<div class="figure align-default" id="directive-fig">
<a class="reference internal image-reference" href="../_images/C-3PO_droid.png"><img alt="../_images/C-3PO_droid.png" src="../_images/C-3PO_droid.png" style="height: 150px;"></a>
<p class="caption"><span class="caption-number">Fig. 4 </span><span class="caption-text">Here is my figure caption!</span><a class="headerlink" href="#directive-fig" title="Permalink to this image">¶</a></p>
</div>
```

If that's not a11y-friendly I'm happy to just go with alt tags

Reply:
My recommendation is that we remove the alt text of the diagrams and review them. By remove, I mean set the alt text to empty. By review, I mean we should make sure that the surrounding text describes the diagram. In other words, we should make sure the diagrams do not contain any information that isn't already conveyed by reading the article without the diagrams. I want to be clear that I am only referring to diagram-type images, not all images.

Here's why. 

I did a deep dive into this subject this afternoon. I dug through threads on the global accessibility Slack. I read web pages on the W3C and Gov.UK sites. I also tested with my own hardware using combinations of VoiceOver and NVDA with Safari, Chrome and Edge. 

The main problem is that there are cross-screen-reader compatibility issues. Not all screen readers treat the figure tag the same way. 

The other problem is that my alt-text diagram descriptions typically repeat or paraphrase stuff that's already in the article. So unless we make my descriptions redundant/repetitive for all users, I don't think we should potentially degrade the screen reader experience by making it more repetitive and verbose.

This seems to align with the [UK government's guidance for describing images.](https://www.gov.uk/guidance/content-design/images)

If we agree with this course of action, then I will review the PR and use the suggestion feature to make the changes I that I am recommending.

Reply:
I defer to your and others expertise when it comes to the vest practices for accessibility. Though it sounds like some of the changes you suggest would also just be more generally useful for readers, if we are removing duplicate information and such. So I'm extra +1 on that as well.